docs/solutions/logic-errors/session-save-format-consistency-2026-05-05.md

```
@@ -0,0 +1,147 @@
+  - session-persistence
+  - serialization
+  - format-consistency
+  - tool-calls
+plan_ref: "fix-437-edit-session-corruption"
+---
+
+## Problem
+
+Running `.edit session` in harnx followed by exiting the editor (even without changes) corrupted session files containing tool-call history. The session became permanently unreadable with error:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)